src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
+      | `Quick_run
+      | `Spark of string list (* custom parameters *)
+      | `Custom of string
+    ] [@@deriving yojson, show]
+  end
```

Comment:
They can, but then it won't be a `Requirement.t`, specifically one won't be able to pass them to `with_requirements`